connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryOffsetBackingStore.java

```diff
@@ -52,11 +52,15 @@ public void configure(WorkerConfig config) {
 
     @Override
     public synchronized void start() {
+        executor = Executors.newSingleThreadExecutor();
     }
 
     @Override
     public synchronized void stop() {
-        // Nothing to do since this doesn't maintain any outstanding connections/data
+        if (executor != null) {
+            executor.shutdown();
+            executor = null;
```

Comment:
OK, 30s it is.  For comparison, `KafkaOffsetBackingStore.stop()` uses a `KafkaBasedLog` which ultimately calls `Thread.join()` with *no timeout*, and throws a `ConnectException` if it is interrupted; but there are other paths where it just logs shutdown failures.